ouroboros-consensus/tools/db-convert/Main.hs

```diff
@@ -0,0 +1,220 @@
+              configFile
+              genesisHash
+        case econfig of
+          Left err -> throwIO $ MkConfigError err
+          Right config -> validateChainDb dbDir' config verbose
+
+convertEpochFile
+  :: EpochSlots
+  -> Path Abs File -- ^ Input
+  -> Path Abs Dir -- ^ Ouput directory
+  -> IO (Either CC.ParseError ())
+convertEpochFile es inFile outDir =
+  let inStream = CC.parseEpochFileWithBoundary es (toFilePath inFile)
+      dbDir = outDir </> [reldir|immutable|]
+      encode = CB.serializeEncoding' . Byron.encodeByronBlock . Byron.ByronBlockOrEBB
+   in do
+        createDirIfMissing True dbDir
+        -- Old filename format is XXXXX.dat, new is epoch-XXX.dat
+        outFileName <-
+          parseRelFile $ "epoch-" <>
+            drop 2 (toFilePath (filename inFile))
+        outFile <- (dbDir </> outFileName) -<.> "dat"
+        IO.withFile (toFilePath outFile) IO.WriteMode $ \h ->
+          runResourceT $ runExceptT $ S.mapM_ (liftIO . BS.hPut h) . S.map encode $ inStream
+
+validateChainDb
+  :: forall blk. (blk ~ ByronBlockOrEBB ByronConfig)
+  => Path Abs Dir -- ^ DB directory
+  -> CC.Genesis.Config
+  -> Bool -- Verbose
+  -> IO ()
+validateChainDb dbDir cfg verbose =
+  withThreadRegistry $ \registry -> do
+    chaindb <- give protocolMagicId $ give epochSlots $ ChainDB.openDB $ args registry
+    blk <- ChainDB.getTipBlock chaindb
+    putStrLn $ "DB tip: " ++ condense blk
+    ChainDB.closeDB chaindb
+  where
+    byronProtocolInfo =
+      protocolInfoByron
+        cfg
+        (Just $ PBftSignatureThreshold 0.22) -- PBFT signature threshold
+        (CC.Update.ProtocolVersion 1 0 0)
+        ( CC.Update.SoftwareVersion
+          (CC.Update.ApplicationName "Cardano SL")
+          2
+        )
+        Nothing
+    protocolMagicId = CB.unAnnotated . getAProtocolMagicId $ CC.Genesis.configProtocolMagic cfg
+    epochSlots = CC.Genesis.configEpochSlots cfg
+    securityParam = SecurityParam $ CC.unBlockCount k
+    k = CC.Genesis.configK cfg
+    args registry =
+      (ChainDB.defaultArgs @blk (toFilePath dbDir))
+        { ChainDB.cdbGenesis = return $ pInfoInitLedger byronProtocolInfo
+        , ChainDB.cdbDecodeBlock = Byron.decodeByronBlock epochSlots
+        , ChainDB.cdbDecodeChainState = Byron.decodeByronChainState
+        , ChainDB.cdbDecodeHash = Byron.decodeByronHeaderHash
+        , ChainDB.cdbDecodeLedger = Byron.decodeByronLedgerState
+        , ChainDB.cdbEncodeBlock = Byron.encodeByronBlock
+        , ChainDB.cdbEncodeChainState = Byron.encodeByronChainState
+        , ChainDB.cdbEncodeHash = Byron.encodeByronHeaderHash
+        , ChainDB.cdbEncodeLedger = Byron.encodeByronLedgerState
+        , -- Policy
+        ChainDB.cdbValidation = ImmDB.ValidateAllEpochs
+        , ChainDB.cdbBlocksPerFile = 10
+        , ChainDB.cdbMemPolicy = LgrDB.defaultMemPolicy securityParam
+        , ChainDB.cdbDiskPolicy = LgrDB.defaultDiskPolicy securityParam 20000
+        , -- Integration
```

Comment:
Can you fix this line too? And maybe run `stylish-haskell` for uniformity's sake :slightly_smiling_face:

Reply:
`stylish-haskell` still relies on a version which isn't in nixpkgs, and I'm too stubborn to install...